src/app/components/Labels/NewLabelForm.js

```diff
@@ -0,0 +1,70 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { c } from 'ttag';
+import { Input, Label, Icon } from 'react-components';
+import { LABEL_COLORS } from 'proton-shared/lib/constants';
+
+import LabelColors from './LabelColors';
+
+function randomIntFromInterval(min, max) {
+    return Math.floor(Math.random() * (max - min + 1) + min);
+}
+
+function NewLabelForm({ onSubmit, type, label, onUpdate }) {
```

Comment:
This component should be considered as a pure component and be controlled by his parent (the modal container). Putting responsibility on children is not a good idea when the view has to change.

Here the component is handling to much thing:
- `useEffect`
- `const Exclusive = +(type === 'folder');`
- `Color: LABEL_COLORS[randomIntFromInterval(0, LABEL_COLORS.length - 1)]`

Just forward the changes to his parent.